src/main/scala/com/fulcrumgenomics/alignment/Aligner.scala

```diff
@@ -402,6 +455,87 @@ class Aligner(val scoringFunction: (Byte,Byte) => Int,
     Alignment(query=query, target=target, queryStart=curI+1, targetStart=curJ+1, cigar=Cigar(elems.reverse), score=score)
   }
 
+  /**
+    * Finds the matrix location of the single best alignment from which to backtrack and generate the alignment.
+    * When there are multiple equally best alignments the choice of which one to return is arbitrary.
+    *
+    * @param query the query sequence
+    * @param target the target sequence
+    * @param matrices the alignment matrices to search
+    */
+  private def findBest(matrices: Array[AlignmentMatrix]): MatrixLocation = {
+    // Find location based on alignment mode:
+    //   - start at the end of query and target for Global
+    //   - the highest scoring cell in the last row for Glocal
+    //   - the highest scoring cell anywhere in the matrix for Local
+    val qLen = matrices(0).queryLength
+    val tLen = matrices(0).targetLength
+
+    this.mode match {
+      case Global =>
+        MatrixLocation(qLen, tLen, AllDirections.maxBy { d => matrices(d).scoring(qLen, tLen) })
+      case Glocal =>
+        var (maxScore, maxJ, maxD) = (MinStartScore, -1, Done)
+        forloop(from = 1, until = tLen + 1) { j =>
+          val direction = AllDirections.maxBy(d => matrices(d).scoring(qLen, j))
+          val score = matrices(direction).scoring(qLen, j)
+          if (score > maxScore) {
+            maxScore = score
+            maxJ = j
+            maxD = direction
+          }
+        }
+        MatrixLocation(qLen, maxJ, maxD)
+      case Local =>
+        var (maxScore, maxI, maxJ, maxD) = (MinStartScore, -1, -1, Done)
+        forloop(from = 1, until = qLen + 1) { i =>
+          forloop(from = 1, until = tLen + 1) { j =>
+            val direction = AllDirections.maxBy(d => matrices(d).scoring(i, j))
+            val score = matrices(direction).scoring(i, j)
+            if (score > maxScore) {
+              maxScore = score
+              maxI = i
+              maxJ = j
+              maxD = direction
+            }
+          }
+        }
```

Comment:
just a thought, you could save having to loop over the matrix again if you cached the max-score (and i/j/direction) when filling out the matrix.  The rules would have to be different for each mode, but you that wouldn't be hard to optimize up front.

Reply:
That's a good point.  I think it's likely only valuable for Local though as the other two modes' searches are very limited.  I think I'll leave that alone for now and we can do it later if we like.